idaes/models/costing/SSLW.py

```diff
@@ -1016,13 +1016,21 @@ def cost_compressor(
             integer: whether the number of units should be constrained to be
                 an integer or not (default = True).
         """
-        # Confirm that unit is a turbine
+        # Confirm that unit is a compressor
         if not blk.unit_model.config.compressor:
             raise TypeError(
                 "cost_compressor method is only appropriate for "
                 "pressure changers with the compressor argument "
                 "equal to True."
             )
+        # compressor = True, and using non-isentropic assumption
+        # (costing not needed)
+        elif hasattr(blk.unit_model.config, "thermodynamic_assumption"):
+            if blk.unit_model.config.thermodynamic_assumption.name != "isentropic":
```

Comment:
I am not sure this is quite comprehensive enough - the firs thing that came to mind is what happens if the user is trying to use this on a Pump? In this case, the error message would be misleading - it is not that it can't be costed, but rather that the user is using the wrong costing method. I think we should do one of tow things:

1. Check more comprehensively for the differ types and return a more useful message.
2. Be more permissive, and only raise an error for those cases that don't work (i.e. fail for isothermal, but allow most others to proceeded perhaps with a warning).

Reply:
That's a good point. This would only be an issue when users pass `cost_compressor` as the costing method for PressureChangers, since the `unit_mapping` will see Pumps and use the correct costing method if one is not provided. I agree with your first suggestion and will make the checks more explicit